Sources/Kafka/RDKafka/RDKafkaClient.swift

```diff
@@ -576,8 +580,10 @@ final class RDKafkaClient: Sendable {
     /// Seek for partitions to consume.
     /// - Parameter topicPartitionList: Pointer to a list of topics + partition pairs.
     func seek(topicPartitionList: RDKafkaTopicPartitionList, timeout: Duration) async throws {
+        assert(timeout >= .zero, "Timeout must be positive")
+
         let doSeek = {
-            topicPartitionList.withListPointer { rd_kafka_seek_partitions(self.kafkaHandle, $0, Int32(timeout.inMilliseconds)) }
+            topicPartitionList.withListPointer { rd_kafka_seek_partitions(self.kafkaHandle, $0, Int32(max(timeout, .zero).inMilliseconds)) }
```

Comment:
I guess this one is also blocking?

Reply:
Yes, it is blocking if has timeout, so there was used `performBlockingCall ` in case if timeout is specified couple of lines below